xmodule/course_module.py

```diff
@@ -259,14 +260,19 @@ def default(self):
         return default
 
 
-def get_available_providers():  # lint-amnesty, pylint: disable=missing-function-docstring
+def get_available_providers(course_key=None):  # lint-amnesty, pylint: disable=missing-function-docstring
```

Comment:
@Zacharis278 modified the function itself, but not every use of this function is able to pass in a course key. I'm not sure this is a great approach either, but let me know what you think

Reply:
What if we were to just gate this in the ProctoredExamSettingsView? I don't necessarily see a problem with 'lti_external' always being considered an available provider down at this level and having the waffle gate be more superficial where it just removes the option from UI operations. It might avoid some weirdness if the flag and a course settings got out of sync.

Reply:
I don't think that's a great idea, because of this piece of code: https://github.com/openedx/edx-platform/blob/3cd3c60b590a52a4d9a2f9cbe421832c6633fbbf/xmodule/course_module.py#L236-L245

This has no context for course, so LTI external will always be listed as an available option in the error message, even if the course can't configure it as a provider

Reply:
That's why I was thinking of using the waffle flag to add in the LTI option (which is what it's doing now for the post request). I could remove the code I added to the `get_available_providers` function and into the GET request so that the POST and GET are more similar in their gating behavior.

Reply:
ah I see what you mean, I didn't realize those errors were returned outside of advanced settings. This^ approach this sounds good to me tho

Reply:
Either way (either always including LTI as an available option, or trying to only add it in if the waffle flag is enabled) creates some messiness. I've updated the PR to always include LTI as an option, and selectively remove it if the waffle flag is not enabled. I think this approach works best with the normal field validation so that we don't have to go in and change too much about the current process.